### PR TITLE
Fix issues with Robot Framework 6.1

### PR DESCRIPTION
**Pull request by bhirsz**
Not all issues are fixed but the fatal exception is. Remaining issues will be covered by branch https://github.com/MarketSquare/robotframework-robocop/compare/rf61-support or in additional PRs.

### CODE REVIEW

File: docs/releasenotes/3.2.0.rst [hunk 1]
```diff
@@ -110,6 +110,16 @@ Possibly overwritten variables are now detected in more places (#841, #842)
 The ``possible-variable-overwriting`` now detects issues in ``IF`` assignments, embedded variables
 and variable with default values.
 
+Fix for empty-lines-between-sections failing to parse empty section
+-------------------------------------------------------------------
+
+``empty-lines-between-sections`` rule should now work with empty sections with Robot Framework 6.1.
```
Comment (mnojek) [inline docs/releasenotes/3.2.0.rst:116]: Would be good to add the rule id here.

File: docs/releasenotes/3.2.0.rst [hunk 2]
```diff
@@ -110,6 +110,16 @@ Possibly overwritten variables are now detected in more places (#841, #842)
...
+
+Fix for empty-metadata not reported in Robot Framework 6.1
+-----------------------------------------------------------
+
+``empty-metadata`` rule should now work with Robot Framework 6.1.
```
Comment (mnojek) [inline docs/releasenotes/3.2.0.rst:121]: Would be good to add the rule id here.

**Review by mnojek — APPROVED** [review]

**Comment by bhirsz** [comment]
@mnojek I have rebased your branch on top of my changes. I have also fixed several other issues (the goal is to have green build in this PR). Currently only parsing-error tests are failing - I should fix them today.

**Comment by bhirsz** [comment]
Everything is fixed

File: robocop/checkers/errors.py [hunk 3]
```diff
@@ -282,7 +282,13 @@ class ParsingErrorChecker(VisitorChecker):
         "resource": "Resource",
         "variables": "Variables",
     }
-    ignore_errors = ("can only be used inside a loop",)
+    ignore_errors = (
+        "can only be used inside a loop",
+        "is allowed only once. Only the first value is used",
+        "Test name cannot be empty",  # handled by test-case-name-is-empty
+        "User keyword name cannot be empty",  # handled by keyword-name-is-empty
+        "END is not allowed in this contex",  # handled by statement-outside-loop
```
[the remark this answers is not in the document]
Comment (bhirsz) [reply robocop/checkers/errors.py:290]: Ah, it does not - and it works since we are checking "in error" and not equal. I will fix it

**Comment by mnojek** [comment]
Thanks @bhirsz for this work! Massive amount of fixes and improvements 🔥

**Pull request merged**